### ISSUE TITLE
got segment fault if set `hot-region-schedule-limit = 0`

### ISSUE THREAD

**Issue by solotzg** [tikv/pd#4145]
### What did you do?

In https://ci.pingcap.net/blue/organizations/jenkins/tiflash_regression_test_daily/detail/tiflash_regression_test_daily/612/pipeline/
set 
```
[schedule]
hot-region-schedule-limit = 0
```
run with pd

### What did you see instead?
```
[2021-09-22T09:24:58.975Z] panic: runtime error: invalid memory address or nil pointer dereference

[2021-09-22T09:24:58.975Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1af4f0a]

[2021-09-22T09:24:58.975Z] 

[2021-09-22T09:24:58.975Z] goroutine 385 [running]:

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server.(*Handler).PackHistoryHotReadRegions(0xc001d01920, 0x1a93c53, 0xc001916120, 0xc00078e0b0, 0xc002baeea8, 0x1af4e92)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/handler.go:915 +0x4a

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server/core.(*HotRegionStorage).pullHotRegionInfo(0xc000646f80, 0xc002baef01, 0x0)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/core/hot_region_storage.go:222 +0x38

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server/core.(*HotRegionStorage).backgroundFlush(0xc000646f80)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/core/hot_region_storage.go:182 +0x177

[2021-09-22T09:24:58.975Z] created by github.com/tikv/pd/server/core.NewHotRegionsStorage

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/core/hot_region_storage.go:136 +0x21e

[2021-09-22T09:24:58.975Z] [2021/09/22 09:23:57.317 +00:00] [WARN] [retry_interceptor.go:61] ["retrying of unary invoker failed"] [target=endpoint://client-321cbb22-ec41-492f-a113-fab74778de43/127.0.0.1:6492] [attempt=0] [error="rpc error: code = DeadlineExceeded desc = context deadline exceeded"]

[2021-09-22T09:24:58.975Z] panic: runtime error: invalid memory address or nil pointer dereference

[2021-09-22T09:24:58.975Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1935ffe]

[2021-09-22T09:24:58.975Z] 

[2021-09-22T09:24:58.975Z] goroutine 4782 [running]:

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server/election.(*lease).KeepAlive(0x0, 0x311fed0, 0xc0043b3ac0)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/election/lease.go:99 +0x9e

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server/election.(*Leadership).Keep(0xc000432b40, 0x311fed0, 0xc004588180)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/election/leadership.go:131 +0xb7

[2021-09-22T09:24:58.975Z] github.com/tikv/pd/server/member.(*Member).KeepLeader(0xc000823620, 0x311fed0, 0xc004588180)

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/member/member.go:154 +0x45

[2021-09-22T09:24:58.975Z] created by github.com/tikv/pd/server.(*Server).campaignLeader

[2021-09-22T09:24:58.975Z] 	/home/jenkins/agent/workspace/build_pd_multi_branch_master/go/src/github.com/pingcap/pd/server/server.go:1240 +0x71f

```

all log files: http://fileserver.pingcap.net/download/logs/pingcap/tiflash/tiflash-jenkins-test-log-tiflash_regression_test_daily-612.tar.gz

### What version of PD are you using (`pd-server -V`)?
```
[2021-09-22T07:42:49.912Z] => pd

[2021-09-22T07:42:49.912Z]    Release Version: v5.3.0-alpha-24-gef40841f

[2021-09-22T07:42:49.912Z]    Git Commit Hash: ef40841fee7f12c89c7562ae149013fb81a3e42b

[2021-09-22T07:42:49.912Z]    Git Branch: master

[2021-09-22T07:42:49.912Z]    UTC Build Time:  2021-09-15 06:49:50
```

**Comment by rleungx** [comment]
PTAL @qidi1

**Comment by nolouch** [comment]
https://github.com/tikv/pd/pull/4121 will fix this problem.

**Comment by qidi1** [comment]
I think this is already fix in #4121

**Comment by qidi1** [comment]
I think this is already fix in #4121
